### ISSUE TITLE
Provide interacting seed base pairs instead of seed region as input

### ISSUE THREAD

**Issue by michauhl** [BackofenLab/IntaRNA#33]
Provide interacting base pairs as string in command line input. E.g.
-x 2,3pu4p-1234,7p
So the format here would be:
first_bp_position_query,bp_info_string-first_bp_position_target,bp_info_string
NOTICE that for the target the positions go downwards, here bps would be 2-1234,3-1233,4-1232 ...
This would only support 1 query - 1 target input. Otherwise IDs have to be used too, and strings would get pretty long. Might be better to use an input file for more seed definitions. On the other hand, for e.g. 3 interactions 9 seed definitions would have to be supplied (in an all vs all fashion). So this option probably only makes sense for one input pair, provided as sequences in command line (the way I always used IntaRNA1, called inside a script, store IntaRNA output directly in variable).

**Comment by martin-mann** [comment]
since seeds are (supposed to be) relatively short and comprise only a few base pairs, we can use a dot-bracket notation ala

```
start1DB1:DB2end2 
```

so encoding as if they are concatenated at the (right) end of the seed

```
             6    10
             |    |
     5'-GGUUU    N CAUA...AGU-3'
             CC A A
             || | |
             GG U U
       3'- UU  C   CCCCACAAA-5'
             |    |
            14    10
```
is encoded by
```
6(((.(:)).))14
```
what do you think...

**Comment by martin-mann** [comment]
concerning multi-seed inputs: I agree that it would be easiest (and sufficient) to restrict this feature to single query/target input

**Comment by PatrickRWright** [comment]
Since I never use this encoding its hard for me to evaluate the benefits.

**Comment by martin-mann** [comment]
@PatrickRWright this might be a reasonable "one-line-string" encoding for interactions too (in general), which might be interesting for you (to store, load, handle interactions)

**Comment by martin-mann** [comment]
Another idea concerning encoding is to use `|` for interacting base pairs like

```
6|||.|:10||.||
```
this way, one can combine intramolecular and interacting base pair information while keeping it readable. e.g. encoding a kissing hairpin (seed) interaction would look like this:
```
3(((|||.|))):6((((||.||))))
```
another benefit of using `|` is that you can now swap the two substrings without recoding the intermolecular base pairs.

**Comment by michauhl** [comment]
Sounds like a nice extra feature yes. For me the first case would be sufficient. The second case with intramolecular base pairs in the seed region would be only applicable for small stem loops. I cannot think of a use case here right now, but who knows :)

**Comment by martin-mann** [comment]
implemented in dd57b4b6d6ce70e696ee63de79b7e4e737c869f9
